tests/unit/src/lib/validator.test.js

```diff
@@ -77,36 +83,47 @@ describe("should fail all the bad ninos", () => {
   test.each(bad_prefixes_ninos)(
     "given bad prefix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_unused_prefixes_ninos)(
     "given unused prefix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_suffixes_ninos)(
     "given bad suffix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_length_ninos)(
     "given bad length nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
 });
 
+describe("invoke the invalidCharacters function", () => {
+  it("should return formattedValue when isValidNino does not return the same value", () => {
+    const testValue = "abc def";
+    const formattedValue = testValue.replaceAll(" ", "").toUpperCase();
+    const isValidNinoResult = isValidNino(formattedValue);
+    expect(isValidNinoResult).to.not.equal(formattedValue);
+    const result = invalidCharacters(formattedValue);
+    expect(result).to.equal(formattedValue);
```

Comment:
I wish there wasn't two thing being tested here and I sort of get what you are trying to do, the focus on formattedValue as makes the test a little difficult. I think if you created an array with exact test values with examples being test with actual hard coded values it would be better